src/main/java/com/cloudbees/jenkins/plugins/bitbucket/api/BitbucketApi.java

```diff
@@ -239,5 +242,20 @@ boolean checkPathExists(@NonNull String branchOrHash, @NonNull String path)
      * @throws InterruptedException if interrupted while waiting on remote communications.
      */
     boolean isPrivate() throws IOException, InterruptedException;
-
+    
+	/**
+	 * @param parent
+	 * @return
+	 * @throws IOException
+	 * @throws InterruptedException
+	 */
+	public Iterable<SCMFile> getDirectoryContent(BitbucketSCMFile parent) throws IOException, InterruptedException;
```

Comment:
Bakes a dependency between the client and SCM API

Fine, but makes it hard to refactor the api into a separate project (if we ever want to do that)

We need to decide if that is a goal or non goal and flag these methods as `@Restricted` if the plan could include spinning out

Reply:
I do not think thats a goal, BitbucketAPI is specific to branch source implementation, is not interned to be used as generic bitbucket API. Perhaps we can mark `@Restricted` at the `BitbucketAPI` class level.

Reply:
Well extension plugins may need the client api, so I think just restrict these two methods for now